lucene/replicator/src/java/org/apache/lucene/replicator/nrt/CopyOneFile.java

```diff
@@ -96,11 +96,12 @@ public boolean visit() throws IOException {
         // Paranoia: make sure the primary node is not smoking crack, by somehow sending us an already corrupted file whose checksum (in its
         // footer) disagrees with reality:
         long actualChecksumIn = in.readLong();
-        if (actualChecksumIn != checksum) {
+        // CheckSum is written in Big Endian so we need to reverse bytes
+        if (actualChecksumIn != Long.reverseBytes(checksum)) {
```

Comment:
IMO we should instead fix the above call to do `long actualChecksumIn = Long.reverseBytes(in.readLong());` to get the actual checksum value? This way the below error message would also be correct?

Reply:
yes, just need to make sure we revert bytes when writing the checksum again.